daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -407,6 +446,21 @@ class VariableMap private(vTable: Map[GlobalQName, ArrayBuffer[VariableInstance]
         }
 
         case _ => {
+          vrd.direction match {
+            /**
+             * Due to potential race conditions regarding the setting of
+             * variables via setVariable and default values in cominbation with
+             * suspesions during unparsing, we only allow the use of either
+             * setVariable statements or a default value when unparsing a
+             * variable.
+             */
+            case VariableDirection.UnparseOnly | VariableDirection.Both if (vrd.maybeDefaultValueExpr.isDefined && abuf.get.size > 1) => {
```

Comment:
abuf is the structure that accumulates the related NVIs for this variable? 

This is fine, I'm just wondering if the name "abuf" can be improved. 

How many of these warnings do we get anyway? (None I would guess, except from tests designed to cause it?

If so then I'd suggest making this an SDE. That's a more conservative design point because we can always weaken it to a warning, but once people start depending on schemas working where it's a warning we would break their schema by changing it to an SDE.

Reply:
We had one existing case that caused the warning.  I'm going to create an additional test case specifically to trigger this now that it will be an SDE as the existing test case does provide important scoping testing for newVariableInstances.